php/PHPCD.php

```diff
@@ -1019,4 +1025,68 @@ public function classmap($pattern)
             ];
         }, array_unique($classmap));
     }
+
+    /**
+     * Add support for laravel
+     *
+     * @author: xiaohui.lam
+     */
+    private function addSupportForLaravel() {
+        $composer_path = $this->root . '/composer.json';
+
+        if (!is_readable($composer_path)) {
+            return [];
+        }
+
+        $composer = json_decode(file_get_contents($composer_path), true);
+        foreach(isset($composer['require'])?$composer['require']:[] as $package => $ver) {
+            if($package == 'laravel/framework') {
+                // 有使用 laravel 框架
+                // 增加 facades 支持
+                if(file_exists($this->root.'/config/app.php')) {
+                    $config = require_once($this->root.'/config/app.php');
+                    $aliases = $config['aliases'];
+                    $require = [
+                        'App' => 'Illuminate\Foundation\Application',
```

Comment:
maybe we need a more generic alias support. not only for laravel framework but also for any aliased.

Reply:
Laravel's facades is complex,
For example, DB might aliased to Illuminate\Database\DatabaseManager::class, but also can aliased to Illuminate\Database\Connection::class (but laravel provides a hidden class named Illuminate\Database\Capsule\Manager. [patch](https://github.com/lvht/phpcd.vim/pull/105/commits/2ad962902f3f182e80c0f99b146d4b0607fc29bc) ).

So, it's so hard to make a universal formula to revert facaded classes, even we made, it's hard to adapt all frameworks.